README.md

```diff
@@ -39,4 +39,107 @@ TODO
+```rust
+pub trait ExtractDestinationData {
+	fn extract_dest(dest: &MultiLocation) -> Option<(Vec<u8>, DomainID)>;
+}
+```
+
+As a developer who needs to construct `dest: MultiLocation` and then call `deposit`, you need to know how this MultiLocation is structured, for example:
+```
+(parents: 0, interior: X2(GeneralKey("ethereum recipient"), GeneralIndex(destDomainID)))
+```
+
+### MultiAsset
+`MultiAsset` is also a substrate type. Asset can be divided into different types from different point of view, such as fungible and non-fungible assets, 
+native asset and foreign asset ,etc. `MultiAsset` is the concept to handler multiple assets in the Polkadot world.
+In sygma pallets, `MultiAsset` is used to identify the asset no matter its location and fungibility. Below is the definition of `MultiAsset`:
+```rust
+pub struct MultiAsset {
+	pub id: AssetId,
+	pub fun: Fungibility,
+}
+```
+the `AssetID` and `Fungibility` type are defined as:
+```rust
+pub enum AssetId {
+	Concrete(MultiLocation),
+	Abstract(Vec<u8>),
+}
+
+pub enum Fungibility {
+    Fungible(#[codec(compact)] u128),
+    NonFungible(AssetInstance),
+}
+```
+the assetID is a `MultiLocation` and fungibility contains `u128` which is the asset amount if it's a fungible asset.
+
+As a developer who needs to construct `asset: MultiAsset` and then call `deposit`, you need to know how this MultiAsset is structured, for example, the fungible testing asset USDC can be constructed like:
+```rust
+(
+    Concrete(
+        MultiLocation::new(1, X3(Parachain(2004), GeneralKey("sygma"), GeneralKey("usdc")))
+    ), 
+    Fungible(amount)
+)
+```
+
+### DomainID & ChainID
+In sygma pallets, multiple destination domain is supported in one single pallet instance. There are DestDomain management extrinsics to register/unregister domainID with its corresponding ChainID.
+This information is stored in the chain storage:
+```rust
+	pub type DestChainIds<T: Config> = StorageMap<_, Twox64Concat, DomainID, ChainID>;
+```
+
+ChainID is not explicitly used in the pallet logic, but they are registered with DomainID. By querying the getter method of `dest_chain_ids`, it would be easy to find out which domainID is binding with which chainID.
+
+### ResourceID
+ResourceID the identifier of the asset in sygma system. To link it with XCM asset, there is `ResourcePairs` defined in the runtime which is the mapping between `AssetId` and `ResourceID`.
+```rust
+type ResourcePairs: Get<Vec<(AssetId, ResourceId)>>;
+```
+As mentioned in the `MultiAsset` section, the `AssetId` contains the asset's MultiLocation, so that one asset with its `MultiLocation` is able to link with `ResouceID`, 
+
+### AssetId of u32 vs AssetId of xcm
+**AssetID of u32**: When creating an asset in substrate, a `u32` number has to be assigned as the `assetID` and this number will be displayed in the assets
```

Comment:
It's not necessary to use u32 to represent asset id, also can use u8 or u128, etc.
The idea of we have an `AssetID of u32` is because we use pallet-assets to manage foreign asset in the example runtime. Developers can config their own implementation, for example, some parachains use orml token to do this, which will have its own concept. Sygma bridge only interacts with asset location and resourceId.

So my point is we  should make developers realize that we just use pallet-assets for demo purposes, they can have their own config.

Reply:
you are absolutely correct, I will add more explanation about this.